conans/client/graph/proxy.py

```diff
@@ -152,12 +152,15 @@ def _retrieve_from_remote(the_remote):
         if not remotes:
             raise ConanException("No remote defined")
         for remote in remotes:
-            try:
-                new_ref = _retrieve_from_remote(remote)
-                return remote, new_ref
-            # If not found continue with the next, else raise
-            except NotFoundException:
+            if remote.disabled:
```

Comment:
Next statement should be `continue` instead of `pass`, and then there is no need to for the `else`.